src/storage/page_buffer.c

```diff
@@ -7345,38 +7372,78 @@ pgbuf_lock_page (THREAD_ENTRY * thread_p, PGBUF_BUFFER_HASH * hash_anchor, const
 			{
 			  prev_thrd_entry->next_wait_thrd = thrd_entry->next_wait_thrd;
 			}
-
 		      thrd_entry->next_wait_thrd = NULL;
-		      pthread_mutex_unlock (&hash_anchor->hash_mutex);
 
-		      perfmon_inc_stat (thread_p, PSTAT_LK_NUM_WAITED_ON_PAGES);	/* monitoring */
-		      return PGBUF_LOCK_WAITER;
+		      break;
 		    }
 		  prev_thrd_entry = thrd_entry;
 		  thrd_entry = thrd_entry->next_wait_thrd;
 		}
-	      pthread_mutex_unlock (&hash_anchor->hash_mutex);
+	      (void) pthread_mutex_unlock (&hash_anchor->hash_mutex);
 	    }
 	  perfmon_inc_stat (thread_p, PSTAT_LK_NUM_WAITED_ON_PAGES);	/* monitoring */
-	  return PGBUF_LOCK_WAITER;
+	  pgbuf_lock_state = PGBUF_LOCK_WAITER;
+	  break;
 	}
       cur_buffer_lock = cur_buffer_lock->lock_next;
     }
 
-  /* buf_lock_table is implemented to have one entry for each thread. At first design, it had one entry for each
-   * thread. cur_thrd_entry->index : thread entry index cur_thrd_entry->tran_index : transaction entry index */
+  if (pgbuf_lock_state != PGBUF_LOCK_WAITER)
```

Comment:
Move code to separate function to improve readability. Change the condition a bit, either by specifically setting pgbuf_lock_state to PGBUF_LOCK_HOLDER, or by checking that cur_buffer_lock does not exist.
```suggestion
  if (pgbuf_lock_state == PGBUF_LOCK_HOLDER) // or cur_buffer_lock == nullptr
    {
      cur_buffer_lock = &(pgbuf_Pool.buf_lock_table[thread_p->index]);
      pgbuf_init_buffer_lock (cur_buffer_lock, ...);
    }
```

Reply:
Moved the initialization code only to a separate function. I think that moving too much of the locking code to separate scopes will have the effect of losing context rather than improve it.